### ISSUE TITLE
Auto bisect for broken periodic CI jobs

### ISSUE THREAD

**Issue by huydhn** [pytorch/pytorch#116539]
When a periodic job starts failing in trunk, OSS CI oncall needs to figure out which commit breaks it by doing manual bisect on a range of trunk commits before being able to reach out to the author of the change for a mitigation plan.  The whole process is time consuming preventing oncall from doing more useful things.

Inspired by internal toolings, the proposal here is to make the process on GitHub more automated:

* **Bisect**.  Automatically trigger the missing periodic job to gather signals
* **Notify**.  Once identified, cc the author of the PR about the failure with link and information automatically via GitHub bot
* **Revert**.  We have several choices here to either submit a revert PR like DK, or to revert the commit automatically

**Comment by xuzhao9** [comment]
Maybe we could reuse the existing Torchbench bisector: https://github.com/pytorch/benchmark/blob/main/bisection.py, it has been working pretty well for Torchbench tests.

**Comment by huydhn** [comment]
So far, oncalls have been doing this manually by pushing `ciflow/periodic` on any suspicious trunk commits, wait for the workflow to finish, and gather the signals.  That's not too bad, it could be better I guess.

**Comment by xuzhao9** [comment]
@huydhn Yeah one simple optimization is to only run the regressed tests on the suspicious trunk commits, that will save a lot of time since common cases are most tests are unaffected.
